File: ruby/pe009.rb
#!/usr/bin/env ruby
# http://projecteuler.net/problem=9

(1..332).each do |a|
  (a..(999 - a) / 2).each do |b|
    c = 1000 - b - a
    next unless a ** 2 + b ** 2 == c ** 2
    puts a * b * c
    exit
  end
end

File: go/pe009.go
// http://projecteuler.net/problem=9
package main

import "fmt"

func main() {
	for a := 1; a <= 1000; a++ {
		for b := a; b <= 1000; b++ {
			c := 1000 - b - a

			if a*a+b*b == c*c {
				fmt.Println(a * b * c)
				return
			}
		}
	}
}
